client/my-sites/hosting/sftp-card/ssh-keys.tsx

```diff
@@ -0,0 +1,110 @@
+import { useSSHKeyQuery } from 'calypso/me/security-ssh-key/use-ssh-key-query';
+import { getCurrentUserId } from 'calypso/state/current-user/selectors';
+import SshKeyCard from './ssh-key-card';
+import { useAtomicSshKeys } from './use-atomic-ssh-keys';
+import { useAttachSshKeyMutation } from './use-attach-ssh-key';
+
+import './ssh-keys.scss';
+
+interface SshKeysProps {
+	siteId: number;
+	userId: number | null;
+	translate: LocalizeProps[ 'translate' ];
+	disabled: boolean;
+}
+
+function SshKeys( { translate, siteId, userId, disabled }: SshKeysProps ) {
+	const { data: keys, isLoading: isLoadingKeys } = useAtomicSshKeys( siteId, {
+		enabled: ! disabled,
+	} );
+	const { data: userKeys, isLoading: isLoadingUserKeys } = useSSHKeyQuery();
+	const { mutate: attachSshKey, isLoading: attachingKey } = useAttachSshKeyMutation( siteId );
+
+	const [ selectedKey, setSelectedKey ] = useState( 'default' );
+	function onChangeSelectedKey( event: React.ChangeEvent< HTMLSelectElement > ) {
+		setSelectedKey( event.target.value );
+	}
+
+	const userKeyIsAttached = useMemo( () => {
+		if ( ! userKeys || ! keys ) {
+			return false;
+		}
+		let foundKey = false;
+		const siteFigerprints = keys.map( ( { fingerprint } ) => fingerprint );
+		for ( const key of userKeys ) {
+			if ( siteFigerprints.includes( key.sha256 ) ) {
+				foundKey = true;
+			}
+		}
+		return foundKey;
+	}, [ keys, userKeys ] );
+
+	const isLoading = isLoadingKeys || isLoadingUserKeys;
+	const showKeysSelect = ! isLoading && ! userKeyIsAttached && userKeys && userKeys.length > 0;
+	const showLinkToAddUserKey = ! isLoading && ! userKeyIsAttached && userKeys?.length === 0;
+
+	return (
+		<div className="ssh-keys">
+			<label htmlFor="attach-ssh-key" className="form-label">
+				{ translate( 'SSH Keys' ) }
+			</label>
+
+			{ keys?.map( ( { name, fingerprint } ) => (
+				<SshKeyCard
+					key={ fingerprint }
+					name={ name }
+					fingerprint={ fingerprint }
+					deleteText={ translate( 'Delete' ) }
+					userId={ userId }
+					siteId={ siteId }
+				/>
+			) ) }
+
+			{ isLoading && <Spinner /> }
+
+			{ showKeysSelect && (
+				<FormFieldset className="ssh-keys__form">
+					<FormSelect
+						id="attach-ssh-key"
+						className="ssh-keys__select"
+						onChange={ onChangeSelectedKey }
+						value={ selectedKey }
+					>
+						{ userKeys?.map( ( key ) => (
+							<option value={ key.name } key={ key.sha256 }>
+								{ key.name }
+							</option>
+						) ) }
+					</FormSelect>
+					<Button
+						primary
+						onClick={ () => attachSshKey( { name: selectedKey } ) }
+						disabled={ attachingKey }
+						busy={ attachingKey }
+					>
+						<span>{ translate( 'Attach SSH key to site' ) }</span>
+					</Button>
+				</FormFieldset>
+			) }
+			{ showLinkToAddUserKey && (
+				<div>
+					<a href="/me/security/ssh-key">{ translate( 'Add an SSH key to your account' ) }</a>{ ' ' }
+					{ translate( 'in order to use it with this site.' ) }
```

Comment:
I think we might want to use `createInterpolateElement` so we can keep this as one string instead of two:

https://github.com/Automattic/wp-calypso/blob/8b16f5d780c6bf7540f273bd7928d9b7d06545b6/client/me/security-ssh-key/security-ssh-key.tsx#L119-L135

Two separate strings will mean the `in order to use it with this site.` will be translated incorrectly (because it's missing the first part of the sentence as context)

Reply:
Done! 
Wow!, I'll be addicted to `createInterpolateElement` so useful !!! 😍 .